SQL-assignment.sql

```diff
@@ -0,0 +1,70 @@
+use baseball;
+
+# create a table that has a column called DATEFIDD whcih calculate
+# the differecnes day between a specific date and local_date in game table
+drop table if exists batting_avg_wDATEDIFF ;
+CREATE table batting_avg_wDATEDIFF (
+batter int unsigned not null,
+hit int unsigned not null,
+atbat int unsigned not null,
+local_date datetime not null,
+diff_day int not null
+) engine=InnoDB default charset latin1;
+
+# Insert data into batting_avg_wDATEDIFF table
+insert into batting_avg_wDATEDIFF (
+batter,hit,atbat,local_date,diff_day)
+select bc.batter,bc.hit,bc.atBat,g.local_date,DATEDIFF('2012-07-05 00:00:00' ,g.local_date) 
```

Comment:
DATEDIFF is a cool way to solve this that I wish I thought of. But you should find a way to remove negative dates as well. Otherwise games that happen in the future of the date you specify are being included later.
(Looks like you're working with a date near the end of the database where you didn't hit this problem. But running it for a date in 2010 gave me a bunch of negative numbers in the table.)

Reply:
the datadiff function calculates the different days between first expression and second expression and the date I selected is the date a little after latest local_date so there is no negative date issue!